### PR TITLE
Added includeNewOnMatch option to include custom option when an exact…

### PR DESCRIPTION
**Pull request by csuich2**
… match is found

### CODE REVIEW

**Comment by ericgio** [comment]
Yeah, I think both the `filterBy` or `labelKey` props could be good models. One difference is that `addCustomOption` currently returns the set of results, either with or without the custom option, so it may need to be changed to something like `shouldAddCustomOption`, which would be replaced by the callback. Hope that makes sense. Thanks for taking a crack at this.

**Comment by csuich2** [comment]
I took a stab at it, making allowNew be a function that returns true or false if the new entry should be included. If you were thinking of a different signature or contract on the allowNew prop, that isn't a problem.

I didn't rename to shouldAddCustomOption since addCustomOption *already* returns an unmodified collection if the trimmed text is falsey or if there is an exact match. If you want it renamed, that can  definitely be done.

File: src/utils/addCustomOption.js [hunk 1]
```diff
@@ -1,15 +1,28 @@
 import {uniqueId} from 'lodash';
 import {getOptionLabel, getStringLabelKey} from './index';
 
-function addCustomOption(results, text, labelKey) {
-  const exactMatchFound = results.some((o) => (
-    getOptionLabel(o, labelKey) === text
-  ));
-
-  if (!text.trim() || exactMatchFound) {
+function addCustomOption(results, text, labelKey, props) {
```
Comment (ericgio) [inline src/utils/addCustomOption.js]: Given that `text` and `labelKey` are part of `props`, I think it makes sense to just collapse the signature: 

```jsx
addCustomOption(results, props) { ... }
```

File: src/utils/addCustomOption.js [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+      getOptionLabel(o, labelKey) === text
+    ));
+
+    if (exactMatchFound) {
+      return results;
+    }
+  } else {
+    // Otherwise let the allowNew function determine
+    // whether the new entry should be added
+    if (!props.allowNew(results, text, labelKey)) {
```
Comment (ericgio) [inline src/utils/addCustomOption.js]: With the same signature here.

**Comment by ericgio** [comment]
Thanks @csuich2, this looks great. I appreciate the thorough PR.

**Comment by ericgio** [comment]
Released in v3.2.0

**Comment by csuich2** [comment]
👍 My pleasure! Thanks for your help & being so receptive. Great work on this component, btw!

**Pull request merged**